### PR TITLE
wrapped in try-catch failing parts

### PR DESCRIPTION
**Pull request by alexgubanow**

### CODE REVIEW

File: codeviz.py [hunk 1]
```diff
@@ -29,16 +29,16 @@ def __init__(self, filename):
...
-        return includes
+        try:
+            with open(filename, 'rt') as f:
+                    data = f.read()
+            # Remove all comments
+            data = re.sub(r'/\*.*?\*/', '', data, flags=re.DOTALL)
+            data = re.sub(r'//.*', '', data)
+            return includes_re.findall(data)
+        except UnicodeError as ex:
+                print(f"Cann't read file: \"{filename}\", position: {ex.start}")
```
Comment (jmarkowski) [inline codeviz.py]: Typo ("Cann't"), also use single quote strings. Best not to use contractions. Instead, "Cannot read file".

File: codeviz.py [hunk 2]
```diff
@@ -29,16 +29,16 @@ def __init__(self, filename):
...
-        # Remove all comments
-        data = re.sub(r'/\*.*?\*/', '', data, flags=re.DOTALL)
-        data = re.sub(r'//.*', '', data)
-
-        includes = includes_re.findall(data)
-
-        return includes
+        try:
+            with open(filename, 'rt') as f:
+                    data = f.read()
```
Comment (jmarkowski) [inline codeviz.py:34]: Please fix indentation, here, and line 40.

Also, it appears that the only part that needs to wrapped in the try-catch is the file open operation. 

Wrap the try statement around the data reading, and `return []` at the exception with the message.

File: codeviz.py [hunk 3]
```diff
@@ -146,16 +146,19 @@ def create_dot_file(nodes, edges):
...
+                        if n.highlight:
+                            f.write('    node [fillcolor="#ff9999", style=filled]')
+                        else:
+                            f.write('    node [fillcolor="#ff9999", style=filled]')
+                    elif n.filetype == 'header':
+                        if n.highlight:
+                            f.write('    node [fillcolor="#ccffcc", style=filled]')
+                        else:
+                            f.write('    node [fillcolor="#ccccff", style=filled]')
+                except:
```
Comment (jmarkowski) [inline codeviz.py:160]: What's failing here exactly? It'd be great to catch the exact exception, instead of a generic `except` case.

**Comment by alexgubanow** [comment]
ok, lets make it clear, i have found both try catch needed, i can't share files on which i was testing. first is against non ascii chars in files, second is against non recognized filetype, probably you can avoid second one, but i needed working solution, now. Please investigate this as you want, this pull request is just to show problems in your code, not fix your code

**Comment by jmarkowski** [comment]
> ok, lets make it clear, i have found both try catch needed, i can't share files on which i was testing. first is against non ascii chars in files, second is against non recognized filetype, probably you can avoid second one, but i needed working solution, now. Please investigate this as you want, this pull request is just to show problems in your code, not fix your code

**Comment by jmarkowski** [comment]
OK, will fix. Thanks for pointing it out.

**Pull request closed without merging**